### PR TITLE
use etcd cluster name also in apiserver to collect only the matching etcd cluster

### PR DESCRIPTION
**Pull request by rwaffen**
- set etcd cluster name
- use etcd cluster name also in apiserver to collect only the matching etcd cluster
- add some docu strings

### CODE REVIEW

File: manifests/server/etcd.pp [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+#
 class k8s::server::etcd (
   K8s::Ensure $ensure = 'present',
   String[1] $version  = pick($k8s::etcd_version, '3.5.1'),
 
   Boolean $manage_setup             = true,
   Boolean $manage_firewall          = false,
   Boolean $manage_members           = false,
-  String[1] $cluster_name           = 'default',
+  String[1] $cluster_name           = $k8s::etcd_cluster_name,
```
Comment (ananace) [inline manifests/server/etcd.pp:26]: This should be a `pick($k8s::server::etcd_cluster_name, 'default')`, the `k8s` classes might not be included on nodes that are purely etcd servers, so having a fallback on it is necessary.

File: manifests/init.pp [hunk 2]
```diff
@@ -65,6 +65,7 @@
   Stdlib::IP::Address::Nosubnet $api_service_address = k8s::ip_in_cidr($service_cluster_cidr, 'first'),
   K8s::IP_addresses $dns_service_address             = k8s::ip_in_cidr($service_cluster_cidr, 'second'),
   Stdlib::Fqdn $cluster_domain                       = 'cluster.local',
+  String[1] $etcd_cluster_name                       = 'default',
```
Comment (ananace) [inline manifests/init.pp:68]: Not sure if this needs to be on the main `k8s` class, as it's only ever used by the server components.

**Review by ananace — CHANGES_REQUESTED** [review]

File: manifests/init.pp (hunk 2 above)
Comment (rwaffen) [reply manifests/init.pp:68]: in most other modules one can set all data via the main class. so if i want to use k8s i know i can set everything over k8s::something and have not to look in any other class if there might be a parameter i have to set to change something. but this is only design philosophy.

File: manifests/server/etcd.pp (hunk 1 above)
Comment (rwaffen) [reply manifests/server/etcd.pp:26]: will change that.

File: manifests/server/etcd.pp [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
+#
 class k8s::server::etcd (
   K8s::Ensure $ensure = 'present',
   String[1] $version  = pick($k8s::etcd_version, '3.5.1'),
 
   Boolean $manage_setup             = true,
   Boolean $manage_firewall          = false,
   Boolean $manage_members           = false,
-  String[1] $cluster_name           = 'default',
+  String[1] $cluster_name           = pick($k8s::server::etcd_cluster_name, 'default'),
```
Comment (bastelfreak) [inline manifests/server/etcd.pp:28]: why is the pick required, since `default` is defined as default value?

Comment (rwaffen) [reply manifests/server/etcd.pp:28]: @ananace said in another comment:

> This should be a `pick($k8s::server::etcd_cluster_name, 'default')`, the k8s classes might not be included on nodes that are purely etcd servers, so having a fallback on it is necessary.

**Review by bastelfreak — APPROVED** [review]

**Review by ananace — APPROVED** [review]

**Pull request merged**